src/cli/pack_command.zig

```diff
@@ -835,22 +835,47 @@ pub const PackCommand = struct {
         const bundled_deps = json.get(field) orelse return null;
 
         invalid_field: {
-            var iter = bundled_deps.asArray() orelse switch (bundled_deps.data) {
-                .e_array => return .{},
+            switch (bundled_deps.data) {
+                .e_array => {
+                    var iter = bundled_deps.asArray() orelse return .{};
+
+                    while (iter.next()) |bundled_dep_item| {
+                        const bundled_dep = try bundled_dep_item.asStringCloned(allocator) orelse break :invalid_field;
+                        try deps.append(allocator, .{
+                            .name = bundled_dep,
+                            .from_root_package_json = true,
+                        });
+                    }
+                },
+                .e_boolean => {
+                    const b = bundled_deps.asBool() orelse return .{};
+                    if (!b == true) return .{};
+
+                    if (json.get("dependencies")) |dependencies_expr| {
```

Comment:
npm only bundles dependencies from `"dependencies"`?

Reply:
i was confused too, but pnpm only does `dependencies` https://github.com/pnpm/pnpm/blob/7a9473b2377c0a6da5992e82ae9cb283849844b1/pkg-manager/resolve-dependencies/src/getNonDevWantedDependencies.ts#L17

Reply:
yeah i try manual `npm` again and it only seems to do main `"dependencies"`

Reply:
hmm i just tried it in pnpm, and also same results.

it could be that the feature was added before all the other deps types, or implementation was simple and only did main deps by default for it

what behavior should bun use?

Reply:
let's keep it how it is for now